src/hooks/useGroupedData.ts

```diff
@@ -9,19 +9,34 @@ function useGroupedData<T extends Record<string, unknown>>(
       return [];
     }
 
-    const groupedData = data?.reduce((acc: Record<string, T[]>, item: T) => {
+    const groupedData = data.reduce((acc: Record<string, T[]>, item: T) => {
       const groupKey = item[key];
       if (!acc[groupKey as string]) {
         acc[groupKey as string] = [];
       }
       acc[groupKey as string].push(item);
       return acc;
     }, {});
-    return Object?.entries(groupedData)?.map(([shift, row]) => ({
-      shift,
-      row,
-    }));
+
+    const sortedKeys = Object.keys(groupedData).sort((a, b) => {
+      const order = ["A", "B", "C" , "D"];
+      return order.indexOf(a) - order.indexOf(b);
+    });
+
+    return sortedKeys.map((shift) => {
+      const sortedMembers = groupedData[shift].sort((a, b) => {
+        const dateA = new Date(a.created_at as string).getTime();
```

Comment:
I don't think we'd need type assertion here. Isn't created_at known to be string?

Reply:
Fixed